int64[] shape
float64[] data
